scripts/deploy_argo_workflows.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+set -eo pipefail
+
+echo "Running deployment script with settings:"
+echo "ML_NAMESPACE: $ML_NAMESPACE"
+echo "ML_PIPELINE: $ML_PIPELINE"
+echo "LOCAL_DEPLOYMENT_PATH: $LOCAL_DEPLOYMENT_PATH"
+echo "ENVIRONMENTS_PATH: $ENVIRONMENTS_PATH"
+echo "ENVIRONMENT: $ENVIRONMENT"
+echo "DEPLOYMENT_BRANCH: $DEPLOYMENT_BRANCH"
+
+DEPLOYMENT_REPO=mlplatform-deployments-mlpipeline
+DEPLOYMENT_PATH=workflows/$ML_NAMESPACE/$ML_PIPELINE
+
+echo "Install envsubst & wget"
+apt-get update && apt-get install -y gettext-base wget && apt-get clean
+
+# Note: Argo lint functionality requires access to the kubernetes cluster, so we use kubeval for validation of the
+# deployment files. See https://github.com/argoproj/argo-workflows/issues/2793.
+echo "Install kubeval to validate deployment files"
+wget https://github.com/instrumenta/kubeval/releases/latest/download/kubeval-linux-amd64.tar.gz
+tar xf kubeval-linux-amd64.tar.gz
+cp kubeval /usr/local/bin
+
+echo "Clone the workflow deployment repository"
+export GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no"
+git clone https://github.com/deliveroo/mlplatform-deployments-mlpipeline.git $DEPLOYMENT_REPO
+
+if [ $DEPLOYMENT_BRANCH != "staging" ] && [ $DEPLOYMENT_BRANCH != "main" ]
+then
+  echo "Deployment branch must be either 'staging' or 'main'."
+  exit 1
+fi
+
+cd $DEPLOYMENT_REPO
+git fetch origin $DEPLOYMENT_BRANCH
+git checkout $DEPLOYMENT_BRANCH
+
+echo "Read variables from environment file"
+source $ENVIRONMENTS_PATH/${ENVIRONMENT}
+
+echo "Format list of variables to pass to ensubst"
+ENV_VARS=$(cut -d= -f1 $ENVIRONMENTS_PATH/${ENVIRONMENT})
+export $ENV_VARS
+ENVSUBST_VARS=$(echo $ENV_VARS  | sed 's/[^ ]* */$&/g')
+
+echo "Generate deployment files"
+rm -rf $DEPLOYMENT_PATH
+mkdir -p $DEPLOYMENT_PATH
+for f in $(find $LOCAL_DEPLOYMENT_PATH -regex '.*\.ya*ml'); do envsubst "$ENVSUBST_VARS" < $f > "$DEPLOYMENT_PATH/$(basename $f)"; done
+
+echo "Downloading Argo schema file for schema validation"
+SCHEMA_LOCATION_PATH="/tmp/schema"
```

Comment:
Honestly I'm not sure how (if?) underlying filestystem is shared, but wouldn't it be better to use mktemp here?
```suggestion
SCHEMA_LOCATION_PATH=$(mktemp -d /tmp/schema-XXXXXX)
```

Reply:
Fair point. Let me add that.

Reply:
fixed.